designsystem/src/androidMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/resource/String.kt

```diff
@@ -19,3 +19,10 @@ actual fun provideQuantityStringResource(id: String, quantity: Int): String {
     }
     return provideStringResource("${id}_$appendToId").format(quantity)
 }
+
+@Composable
+actual fun resourceExists(resourceName: String, resourceType: String): Boolean {
+    val context = LocalContext.current
+    val resourceId = context.resources.getIdentifier(resourceName, resourceType, context.packageName)
+    return resourceId != 0
+}
```

Comment:
`@OptIn(ExperimentalResourceApi::class)
@Composable
fun provideDHIS2Icon(resourceName: String, defaultIcon: String = "dhis2_default_outline"): Painter {
    val iconName = if (!resourceExists(resourceName)) {
        defaultIcon
    } else {
        resourceName
    }
    return painterResource("drawable/$iconName.xml")
}`
How about editing the provideDHIS2Icon method to provide default icon when resource is not found. This way we won't have to check in each individual classes

Reply:
good idea! I have just updated it